Gems/ROS2/Code/Source/RobotImporter/URDF/VisualsMaker.cpp

```diff
@@ -231,7 +232,56 @@ namespace ROS2
         entity->Deactivate();
     }
 
-    static void OverrideMaterialPbrSettings(const sdf::Material* material, [[maybe_unused]] AZ::Render::MaterialAssignmentMap& overrides)
+    static void OverrideScriptMaterial(const sdf::Material* material, [[maybe_unused]] AZ::Render::MaterialAssignmentMap& overrides)
+    {
+        AZStd::string materialName(material->ScriptName().c_str(), material->ScriptName().size());
+        if (materialName.empty())
+        {
+            return;
+        }
+
+        // Make sure the material name is lowercased before checking the path in the Asset Cache
+        AZStd::to_lower(materialName);
+        // If a material has a <script> element we'll treat the name as a path and name to an O3DE material.
+        // For example, "Gazebo/Wood" will look for a product material in "<cache>/gazebo/wood.azmaterial"
+        AZ::IO::Path materialProductPath(materialName);
+        materialProductPath.ReplaceExtension(".azmaterial");
+
+        // Try getting an asset ID for the given name.
+        constexpr bool AutoGenerateId = false;
+        AZ::Data::AssetId assetId;
+        AZ::Data::AssetCatalogRequestBus::BroadcastResult(
+            assetId,
+            &AZ::Data::AssetCatalogRequestBus::Events::GetAssetIdByPath,
+            materialProductPath.String().c_str(),
+            azrtti_typeid<AZ::RPI::MaterialAsset>(),
+            AutoGenerateId);
+
+        // No asset was found, we can't convert the material script.
+        if (!assetId.IsValid())
+        {
+            AZ_Warning("AddMaterial", false, "Failed to find product material for %s", materialProductPath.String().c_str());
+            return;
+        }
+
+        // The asset was found, so replace all the material assets in the given material assignment map.
+        AZ::Data::Asset<AZ::RPI::MaterialAsset> materialAsset(assetId, azrtti_typeid<AZ::RPI::MaterialAsset>());
+        for (auto& [id, override] : overrides)
+        {
+            if (id == AZ::Render::DefaultMaterialAssignmentId)
+            {
+                override.m_defaultMaterialAsset = materialAsset;
+            }
+            else
+            {
+                override.m_materialAsset = materialAsset;
+            }
+        }
+
+        AZ_Info("AddMaterial", "Added product material %s", materialProductPath.String().c_str());
```

Comment:
This can use `.c_str()`.
Also we probably want a newline between info messages as `AZ_Info` will not add newlines when the content is written out to the log file.
```suggestion
        AZ_Info("AddMaterial", "Added product material %s\n", materialProductPath.c_str());
```

Reply:
Wait, that's not true. AZ_Info adds newlines.
https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Debug/Trace.h#L325-L329
https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Debug/Trace.h#L114

Reply:
Yeah, that is not the Trace function that runs when running an application like the Editor or Launcher.
That is the interface class that only runs before an `AZ::Debug::Trace` instance is created.

There real function is the [Trace::Printf](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Debug/Trace.cpp#L536-L559) function that runs.
That does not add a newline.

Reply:
Those Trace functions are only used in the O3DE Kernel to log messages before main has run
https://github.com/o3de/o3de/commit/5cec7f916c95e00123d59e145de68b4c25fbbe6e

Reply:
Agh, ty. I put back the newline.